hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -378,19 +379,36 @@ instance MonadIde IdeM where
 
   getPlugins = asks idePlugins
 
+  withIndefiniteProgress t f = do
+    lf <- asks ideEnvLspFuncs
+    withIndefiniteProgress' lf t f
+
 instance MonadIde IdeGhcM where
   getRootPath = lift $ lift getRootPath
   getVirtualFile = lift . lift . getVirtualFile
   getConfig = lift $ lift getConfig
   getClientCapabilities = lift $ lift getClientCapabilities
   getPlugins = lift $ lift getPlugins
+  withIndefiniteProgress t f = do
+    lf <- lift $ lift $ asks ideEnvLspFuncs
+    withIndefiniteProgress' lf t f
 
 instance MonadIde IdeDeferM where
   getRootPath = lift getRootPath
   getVirtualFile = lift . getVirtualFile
   getConfig = lift getConfig
   getClientCapabilities = lift getClientCapabilities
   getPlugins = lift getPlugins
+  withIndefiniteProgress t f = do
+    lf <- lift $ asks ideEnvLspFuncs
+    withIndefiniteProgress' lf t f
+
+withIndefiniteProgress' :: MonadIO m => Maybe (Core.LspFuncs Config) -> T.Text -> m a -> m a
```

Comment:
I couldn't get this to type check with `withIndefiniteProgress = lift . withIndefiniteProgress`, this is a workaround. Open to suggestions to make this nicer